adoc/chapters/device_compiler.adoc

```diff
@@ -387,6 +279,30 @@ literal with greater value.
 In addition, for each <<backend>> supported, the preprocessor macros described
 in <<sec:backends>> must be defined by all conformant implementations.
 
+[[sec:standard-library-support]]
+== {cpp} standard library support
+
+In general, any use of functions or types from the [code]#std# namespace in
+device code produces undefined behavior.
+However, the functions and types listed in this section are guaranteed to have
+defined semantics when used in device code.
+
+=== Type aliases
+
+The following type aliases must alias the same types on host and device:
+
+  * [code]#std::size_t#;
+  * [code]#std::ptrdiff_t#;
+  * [code]#std::max_align_t#;
+  * [code]#std::nullptr_t#; and
```

Comment:
Thinking about it, I was wondering about how this is implemented in C++,  looking at `/usr/include/x86_64-linux-gnu/c++/14/bits/c++config.h` on my laptop, I can see:
```c
  typedef decltype(nullptr)	nullptr_t;
```
which follows https://en.cppreference.com/w/cpp/types/nullptr_t

But how an implementation using for example a different host and device compiler can ensure that these are the same types on host and device side? :thinking: 
Probably at the end, we do not care and the only important aspect is *`sizeof(std::nullptr_t)` is equal to `sizeof(void *)`*.

Reply:
> Probably at the end, we do not care and the only important aspect is _`sizeof(std::nullptr_t)` is equal to `sizeof(void *)`_.

I think we do care that they are the same type on host vs. device.  Otherwise, you could construct constexpr values that are different on host vs. device via `decltype`.

Reply:
It is also important that the types of the type alias targets match so that mangled names can be correlated across host and device compilation. Consider a SYCL kernel name type that is a class template specialization with a `std::size_t` template argument.

Reply:
The problem with `nullptr_t` is that there is a chicken-egg problem since it is defined as the type of `nullptr` and to pass such a value to a device you need to have the device type at the first place, but this is defined from the value...
The mangled name aspect is interesting. Probably the C++ ABI has some mangling for `nullptr_t` too.